tests/unit/test_utils.py

```diff
@@ -0,0 +1,117 @@
+                pool.mkdir(parents=True)
+                archive_root = pool.parent.absolute()
+                distros = [dists / str(i) for i in range(nums)]
+                for distro in distros:
+                    distro.mkdir(parents=True)
+                    index = distro / "Packages"
+                    with index.open(mode="w") as f:
+                        name = "{}.deb".format(str(uuid4))
+                        f.write("Filename: {}\n".format(name))
+                        expected_outputs.add(archive_root / name)
+
+            # get returned output
+            for archive_root, indices in utils.locate_package_indices(tmpdirname):
+                returned_output |= utils.find_packages_by_indices(
+                    indices, base=archive_root
+                )
+
+        self.assertEqual(sorted(expected_outputs), sorted(returned_output))
+
+    def test_locate_package_indices(self):
+        nums = 3
+        expected_outputs = []
+        # Generate a test file structure for testing
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            tmpdirname = pathlib.Path(tmpdirname)
+            for i in range(nums):
+                any_dirs = str(i)
+                dists = tmpdirname / any_dirs / "dists"
+                dists.mkdir(parents=True)
+                pool = tmpdirname / any_dirs / "pool"
+                pool.mkdir(parents=True)
+                archive_root = pool.parent.absolute()
+                distros = [dists / str(i) for i in range(nums)]
+                indices = []
+                for distro in distros:
+                    distro.mkdir(parents=True)
+                    index = distro / "Packages"
+                    index.touch()
+                    indices.append(index)
+                expected_outputs.append((archive_root, indices))
+
+            # get returned output
+            returned_output = sorted(utils.locate_package_indices(tmpdirname))
+
+        for i, outputs in enumerate(returned_output):
+            returned_archive_roots, returned_package_indices = outputs
+            returned_package_indices = sorted(returned_package_indices)
+            self.assertEqual(expected_outputs[i][0], returned_archive_roots)
+            self.assertEqual(expected_outputs[i][1], returned_package_indices)
+
+        non_existing_path = pathlib.Path(str(uuid4()))
+        self.assertRaises(
+            FileNotFoundError, utils.locate_package_indices, non_existing_path
+        )
+
+    def test_convert_bytes(self):
+        units = ["bytes", "KB", "MB", "GB", "TB"]
+        test_params = [
+            (1000 ** (i + 1), "{:.1f} {}".format(1000 ** (i + 1) / 1024 ** (i), unit))
+            for i, unit in zip(range(5), units)
+        ]
+        for num, expected in test_params:
+            with self.subTest():
+                self.assertEqual(utils.convert_bytes(num), expected)
+
+    def test__locate_packages_from_index(self):
```

Comment:
two underline __  ?  I don't know how other people handle this. But I will use only 1 underline. Not a critical issue, just curious.